rc/core/c-family.kak

```diff
@@ -318,34 +318,37 @@ evaluate-commands %sh{
 }
 
 hook global WinSetOption filetype=(c|cpp|objc) %[
-    try %{ # we might be switching from one c-family language to another
-        remove-hooks window c-family-.+
-    }
-
-    hook -group c-family-indent window ModeChange insert:.* c-family-trim-autoindent
-    hook -group c-family-insert window InsertChar \n c-family-insert-on-newline
-    hook -group c-family-indent window InsertChar \n c-family-indent-on-newline
-    hook -group c-family-indent window InsertChar \{ c-family-indent-on-opening-curly-brace
-    hook -group c-family-indent window InsertChar \} c-family-indent-on-closing-curly-brace
-    hook -group c-family-insert window InsertChar \} c-family-insert-on-closing-curly-brace
+    hook -group "%val{hook_param_capture_1}-family-indent" window ModeChange insert:.* c-family-trim-autoindent
+    hook -group "%val{hook_param_capture_1}-family-insert" window InsertChar \n c-family-insert-on-newline
+    hook -group "%val{hook_param_capture_1}-family-indent" window InsertChar \n c-family-indent-on-newline
+    hook -group "%val{hook_param_capture_1}-family-indent" window InsertChar \{ c-family-indent-on-opening-curly-brace
+    hook -group "%val{hook_param_capture_1}-family-indent" window InsertChar \} c-family-indent-on-closing-curly-brace
+    hook -group "%val{hook_param_capture_1}-family-insert" window InsertChar \} c-family-insert-on-closing-curly-brace
 
     alias window alt c-family-alternative-file
-]
-
-hook global WinSetOption filetype=(?!c)(?!cpp)(?!objc).* %[
-    remove-hooks window c-family-.+
 
-    unalias window alt c-family-alternative-file
+    hook -once -always window WinSetOption "filetype=(?!%val{hook_param_capture_1}).*" "
+        remove-hooks window c-family-.+
+    "
+    hook -once -always window WinSetOption filetype=(?!c)(?!cpp)(?!objc).* %{
+        unalias window alt c-family-alternative-file
+    }
 ]
 
-hook -group c-highlight global WinSetOption filetype=c %[ add-highlighter window/c ref c ]
-hook -group c-highlight global WinSetOption filetype=(?!c).* %[ remove-highlighter window/c ]
+hook -group c-highlight global WinSetOption filetype=c %{
+    add-highlighter window/c ref c
+    hook -once -always window WinSetOption filetype=(?!c).* %{ remove-highlighter window/c }
```

Comment:
Using `(?!c).*` is less than ideal, because any filetype that begins with c will not trigger this hook